src/com/amazon/ion/OversizedValueException.java

```diff
@@ -0,0 +1,26 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazon.ion;
+
+/**
+ * An error thrown when the user requests an IonReader to parse a scalar value that
+ * exceeds the reader's configured maximum buffer size. This is never thrown from
+ * {@link IonReader#next()}; oversized values encountered during next() are
```

Comment:
is this true for both incremental and non-incremental modes?

Reply:
Yes, this exception is never thrown from `next()` regardless of mode. Further, this exception is only thrown in non-incremental mode because in incremental mode, all oversized values will be encountered during `next()` due to the fact that the reader attempts to buffer all TLVs during `next()`.